packages/blade-mcp/src/tools/getFigmaToCode.ts

```diff
@@ -0,0 +1,108 @@
+import type { ToolCallback } from '@modelcontextprotocol/sdk/server/mcp.js';
+import dedent from 'dedent';
+import { z } from 'zod';
+import { handleError, sendAnalytics } from '../utils/analyticsUtils.js';
+import { analyticsToolCallEventName } from '../utils/tokens.js';
+
+const URLS = {
+  FIGMA_TO_CODE_URL: {
+    DEV: 'http://localhost:8888',
+    PROD: 'https://blade-chat.dev.razorpay.in',
+  },
+} as const;
+
+const getFigmaToCodeToolName = 'get_figma_to_code';
+
+const getFigmaToCodeToolDescription = `Converts Figma designs into Blade Design System code. Provide a Figma design URL to generate the corresponding React components using Blade's component library.`;
+
+const getFigmaToCodeToolSchema = {
+  fileKey: z
+    .string()
+    .describe(
+      'The unique identifier of your Figma file, found in the URL after "figma.com/file/" or "figma.com/design/". For example, in "figma.com/file/abc123XYZ/MyDesign", the fileKey would be "abc123XYZ". This key is required to access and convert your Figma design.',
+    ),
+  nodeId: z
+    .string()
+    .describe(
+      'The specific element or frame ID within your Figma file that you want to convert. Found in the URL as "node-id=xyz123" when you select an element in Figma. For example, in "figma.com/design/file_id/file_name?node-id=ab-1234", the nodeId would be "ab-1234".',
+    ),
+};
+
+const getFigmaToCodeToolCallback: ToolCallback<typeof getFigmaToCodeToolSchema> = async ({
+  fileKey,
+  nodeId,
+}) => {
+  try {
+    const isProd = process.env.NODE_ENV === 'production';
+    const url = isProd ? URLS.FIGMA_TO_CODE_URL.PROD : URLS.FIGMA_TO_CODE_URL.DEV;
+    const fullUrl = `${url}/figma-to-code`;
+    const response = await fetch(fullUrl, {
```

Comment:
Check if this is working without any additional tls setup

Reply:
tested works.